### ISSUE TITLE
The gate runs the fixer as a critic: normalise worker output before judging it

### ISSUE THREAD

**Issue by AdarGit008** [AdarGit008/mcgyvr#246]
**Blocked by:** nothing. **Blocks:** nothing, but it changes the baseline every
arm below the trunk is measured against, so it wants a round boundary (#231
check 3). **Raised by:** #113's scorer switch, 2026-08-12, from a measurement
taken on real worker output. **Related:** #81 (adjacent and **not** this — see
below), #198 (prompt assembly), #231 (commissioning), ADR-0018, ADR-0019 D3.

## The product runs the fixer as a critic

`ruff` is already a dependency, already installed, already configured, and
already run on every worker output. It is invoked exactly twice, and both times
in judging mode:

```python
# src/mcgyvr/gate/adapters/python.py:69-77
[ruff, "check", "--output-format=json", "--force-exclude", "--", *paths]
# src/mcgyvr/gate/adapters/python.py:117
[ruff, "format", "--diff", "--force-exclude", "--", *paths]
```

Neither `--fix` nor a writing `ruff format` appears anywhere in `src/`. So the
gate **computes the exact diff that would fix the file** — `--diff` literally
produces it — discards it, rejects the change, and then `RetryNotes`
(`src/mcgyvr/escalate.py:185`) hands the findings back to the worker as *"fix
exactly these and change nothing else"*.

**We spend a model call asking a worker to make an edit we already had in hand.**

## What it is worth, measured

Re-scored over the 514 real greedy candidates from the 2026-08-12 null run
(`records/measurements/bench-null-15b-a-2026-08-12/`), `qwen2.5-coder:1.5b`,
scored through `Gate.run`:

| | pass |
|---|---:|
| full gate, raw worker output | 23 / 256 (9.0%) |
| full gate, after `ruff check --fix && ruff format` | **58 / 256 (22.7%)** |
| recovered | **35, with 0 lost** |

**+13.7pp at zero tokens**, 35 gains against 0 losses. That is far past
ADR-0019's `m >= 6` wall — the exact two-sided McNemar p is `2^-35`.

What was rejecting those candidates, none of it a correctness defect:

| count | rule |
|---:|---|
| 61 | blank line contains whitespace |
| 40 | formatter would reflow a worker-added line |
| 27 | `List` vs `list` annotation |
| 9 | import block un-sorted |
| ~10 | line too long |

For scale: the fine-tune (#189) measured **+1.9pp** and cost GPU hours, an export
path and a quantization step. This is a flag.

## This is not #81

#81 routes tasks whose **`task_type`** is `format` / `import_sort` / `lint_fix` /
`rename_symbol` to a program instead of a model — "the task is *format this
file*, so run the formatter." It never touches the output of a
`function_implementation` task, which is what was measured here.

ADR-0019 D3's *"C — cost-only … does not move pass rate. Not a bench lever at
all"* is a claim about **that** rung and is not disturbed by this. This is a
different, unplanned lever: **deterministic normalisation of worker output,
before the gate judges it.** Searched the open and closed issue list on
2026-08-12 — nearest neighbours are #71, #94, #95 — and we did not find one
covering it.

## What this does not touch

The raw completion is captured **before** parsing and scoring
(`tools/breadth/measure.py:578`, then `591`, then `601`). A normalisation step
acts on `parsed.content` on its way into the sandbox, downstream of capture. So
`candidates/*.txt` keep the model's raw bytes, `tools/replies/pin.py` walks those
same raw files, and `tools/finetune/build_dataset.py` trains on them.

**Neither the reply corpus nor the fine-tuning path sees a normalised byte.**
This is a product-behaviour question and nothing else.

## Scope

- **A deterministic normalisation step on worker output, before `Gate.run`.**
  The same binaries the gate already shells to, in their writing mode, with the
  same configuration — no new dependency and no new opinion about what correct
  style is.
- **Decide whether it is unconditional or declared.** A contract whose whole
  point is the file's exact bytes should be able to opt out; `Contract` has no
  such field today.
- **Decide what happens to the retry path.** If normalisation lands, the
  lint/format half of `RetryNotes` becomes mostly dead — a retry spent on a
  finding the fixer would have erased is a wasted model call, and it should stop
  being generated rather than merely stop firing.
- **Report the cost axis.** #113's report carries tokens per condition; this is
  the lever that should move it furthest, and a gain with no cost column is half
  a result (ADR-0018).
- **The JS/TS half.** `prettier` is installed and live on that arm; `eslint` is
  installed and **inert** without a TypeScript parser, and this repository
  carries no eslint configuration at all. Whether the JS side normalises, and
  against which rules, is entangled with that unmade decision.

## Acceptance

- [ ] Worker output is deterministically normalised before the gate judges it,
      using the tools and configuration the gate already uses
- [ ] The step is measured on both outcome axes — acceptance and cost — and the
      cost of the step itself is reported, not assumed to be zero wall-clock
- [ ] A contract can decline normalisation, or the record states why none needs to
- [ ] Retries are no longer generated for findings normalisation would have erased
- [ ] The raw reply corpus is shown unchanged: `pin.py`'s `golden.json` is
      byte-identical across the change

## Out of scope

**Choosing a JS/TS lint standard** — a separate decision about the product's bar
for that language, currently blocking the bench's TypeScript arm. **Changing
which rules the gate rejects on**: this issue makes the product satisfy its
existing bar more cheaply, it does not move the bar.

## The limits of the measurement

- **The Python arm alone.** The TypeScript arm cannot corroborate it while its
  lint rung is inert.
- **Scored by an instrument that is not yet commissioned.** #231 is what would
  license this as a claim in the register rather than an observation in a
  session record. The effect is roughly twice the bench's most pessimistic MDE,
  so it is unlikely to be an artefact of that — but "unlikely" is the honest
  word until the gate passes.
- **One model, one rig, greedy.** srv2, ollama 0.32.5, `qwen2.5-coder:1.5b`.

Re-derive: `records/sessions/lane/113/2026-08-12-the-gate-is-the-scorer-adar.md`.
